sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -191,6 +191,9 @@ private[sql] object JDBCRDD extends Logging {
    */
   private def compileFilter(f: Filter): String = f match {
     case EqualTo(attr, value) => s"$attr = ${compileValue(value)}"
+    case EqualNullSafe(attr, value) =>
+      s"(NOT ($attr != ${compileValue(value)} OR $attr IS NULL OR " +
```

Comment:
It took me several minutes to understand it. Maybe `(attr IS NULL AND value IS NULL) OR (attr <> null AND value <> null AND attr = value)` is easier.